starchiver-ext

```diff
@@ -472,13 +567,101 @@ main_menu() {
     done
 }
 
+function check_for_recent_snapshot() {
+    if [[ "$timedate" == "true" && -z "$snapshot_time" ]]; then
+        echo "Searching for the most recent snapshot file..."
+        local latest_snapshot=$(find "$path" -type f -regex '.*/[0-9]+$' -printf '%T@ %p\n' | sort -r | head -n1 | cut -d" " -f2)
+        if [[ -n "$latest_snapshot" ]]; then
+            # Convert the timestamp to the required format
+            snapshot_time=$(date -d "@$(stat -c %Y "$latest_snapshot")" -d '-1 hour' "+%Y-%m-%d.%H")
+            echo "The adjusted snapshot time is $snapshot_time"
+        else
+            echo "No valid snapshot found within the specified path."
+            exit 1
+        fi
+    fi
+}
+
+function parse_arguments() {
+    while [[ "$#" -gt 0 ]]; do
+        case "$1" in
+            --timedate)
```

Comment:
You might consider naming this "datetime" instead since that's a commonly used term in programming when combining date and time.

Reply:
Oh ok.  Good idea! I will make this change today.

Reply:
That actually makes more sense too. Since that's how the data reads.